### PR TITLE
Add set-as-initiative button to new roll for rerolls

### LINKED ISSUE #10357: Add-To-Initiative button on chat misguides with heropoints
(Sorry for bad title)

When you heropoint a skill check the button to add to initiative from the rolled result becomes grey as-if-inactive and doesn't "move" to the new value, though it does add the new value to initiative
Foundry_Virtual_Tabletop_(21h-55m-07s)  26-09-2023

w/o heropoint:
image

### PR DESCRIPTION
**Pull request by mysurvive**
When rerolling using a hero point, this removes the initiative button from the old roll and re-adds it to the new roll
Resolves #10357

### CODE REVIEW

File: src/module/chat-message/listeners/set-as-initiative.ts [hunk 1]
```diff
@@ -29,7 +29,12 @@ export const SetAsInitiative = {
             setInitiativeButton.title = game.i18n.localize("PF2E.ClickToSetInitiative");
             setInitiativeButton.appendChild(fontAwesomeIcon("fa-swords", { style: "solid" }));
             btnContainer.appendChild(setInitiativeButton);
-            li.querySelector(".dice-total")?.appendChild(btnContainer);
+            if (message.isReroll) {
+                li.querySelector(".pf2e-reroll-discard .dmgBtn-container.dice-total-setInitiative-btn")?.remove();
```
Comment (stwlam) [inline src/module/chat-message/listeners/set-as-initiative.ts:33]: Does anything need to be removed? The set-as-initiative button is added dynamically.

Comment (mysurvive) [reply src/module/chat-message/listeners/set-as-initiative.ts]: Nope, just me overcomplicating it. Thanks!

File: src/module/chat-message/listeners/set-as-initiative.ts [hunk 2]
```diff
@@ -29,7 +29,9 @@ export const SetAsInitiative = {
             setInitiativeButton.title = game.i18n.localize("PF2E.ClickToSetInitiative");
             setInitiativeButton.appendChild(fontAwesomeIcon("fa-swords", { style: "solid" }));
             btnContainer.appendChild(setInitiativeButton);
-            li.querySelector(".dice-total")?.appendChild(btnContainer);
+            message.isReroll
+                ? li.querySelector(".pf2e-reroll-second .dice-total")?.appendChild(btnContainer)
+                : li.querySelector(".dice-total")?.appendChild(btnContainer);
```
Comment (stwlam) [inline src/module/chat-message/listeners/set-as-initiative.ts:34]:
```suggestion
            const selector = message.isReroll ? ".pf2e-reroll-second .dice-total" : ".dice-total"; 
            li.querySelector(selector)?.appendChild(btnContainer);
```

File: src/module/chat-message/listeners/set-as-initiative.ts [hunk 3]
```diff
@@ -29,7 +29,8 @@ export const SetAsInitiative = {
             setInitiativeButton.title = game.i18n.localize("PF2E.ClickToSetInitiative");
             setInitiativeButton.appendChild(fontAwesomeIcon("fa-swords", { style: "solid" }));
             btnContainer.appendChild(setInitiativeButton);
-            li.querySelector(".dice-total")?.appendChild(btnContainer);
+            const selector = message.isReroll ? ".pf2e-reroll-second .dice-total" : ".dice-total"; 
+            li.querySelector(selector)?.appendChild(btnContainer);
```
Comment (stwlam) [inline src/module/chat-message/listeners/set-as-initiative.ts:33]:
```suggestion
            const selector = message.isReroll ? ".pf2e-reroll-second .dice-total" : ".dice-total";
            li.querySelector(selector)?.appendChild(btnContainer);
```

**Pull request merged**